docker/Dockerfile

```diff
@@ -0,0 +1,164 @@
+   && apt-get update \
+   && apt-get install solc
+
+# Install py-solc-x and web3 packages using pip because
+# these are not available in apt repository.
+# Install nose2 for running tests.
+RUN pip3 install --upgrade json-rpc web3 py-solc-x nose2
+
+# Install specific version of solc to support versions <0.6.0
+RUN python3 -m solcx.install v0.5.15
+
+#Build Avalon Enclave Manager intermediate docker image
+FROM base_image as build_image
+
+RUN apt-get update \
+ && apt-get install -y -q \
+    build-essential \
+    pkg-config \
+    cmake \
+    make \
+    libprotobuf-dev \
+    python3-dev \
+    swig \
+    wget \
+    tar
+
+RUN pip3 install --upgrade setuptools
+
+WORKDIR /tmp
+# Build ("Untrusted") OpenSSL, then build ("trusted") SGX OpenSSL
+# Note: The latter will compile in HW or SIM mode depending on the
+# availability of /dev/isgx and /var/run/aesmd/aesm.socket
+RUN OPENSSL_VER=1.1.1d \
+ && wget https://www.openssl.org/source/openssl-$OPENSSL_VER.tar.gz \
+ && tar -zxf openssl-$OPENSSL_VER.tar.gz \
+ && cd openssl-$OPENSSL_VER/ \
+ && ./config \
+ && THREADS=8 \
+ && make -j$THREADS \
+ && make test \
+ && make install -j$THREADS \
+ && ldconfig \
+ && ln -s /etc/ssl/certs/* /usr/local/ssl/certs/ \
+ && cd .. \
+ && rm -rf openssl-$OPENSSL_VER
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./tools/run_tests.sh /project/avalon/tools/
+COPY ./config/tcs_config.toml /project/avalon/config/
+COPY ./common/cpp /project/avalon/common/cpp
+COPY ./common/python /project/avalon/common/python
+COPY ./sdk /project/avalon/sdk
+COPY ./tests /project/avalon/tests
+COPY ./examples/apps /project/avalon/examples/apps
+
+WORKDIR /project/avalon/common/cpp
+
+RUN echo "Building Avalon Common CPP\n" \
+  && mkdir -p build \
+  && cd build \
+  && cmake .. -DUNTRUSTED_ONLY=1 \
+  && make
+
+WORKDIR /project/avalon/common/python
+
+RUN echo "Building Avalon Common Python\n" \
+  && make
+
+WORKDIR /project/avalon/sdk
+
+RUN echo "Building Avalon SDK\n" \
+  && make
+
+# Build Final image and install dependent modules
+FROM base_image as final_image
+
+ARG DISPLAY
+ARG XAUTHORITY
+
+#Environment setup
+ENV http_proxy=$http_proxy
+ENV https_proxy=$https_proxy
+ENV TCF_HOME=/project/avalon
+# Set display, replace value with [IP address of host]:0
+ENV DISPLAY=:0
+
+WORKDIR /project/avalon/
+
+COPY --from=build_image /project/avalon/config/tcs_config.toml /project/avalon/config/
+COPY --from=build_image /project/avalon/examples/apps/ /project/avalon/examples/apps
+COPY --from=build_image /project/avalon/tests /project/avalon/tests
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/tcf_connector.toml /project/avalon/sdk/avalon_sdk/
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/ethereum/contracts/WorkerRegistryList.sol /project/avalon/sdk/avalon_sdk/ethereum/contracts/
+COPY --from=build_image /project/avalon/sdk/avalon_sdk/ethereum/contracts/WorkerRegistry.sol /project/avalon/sdk/avalon_sdk/ethereum/contracts/
```

Comment:
Above two lines of copying contracts can be removed . Were you facing any issue if you don;t copy these files.

Reply:
Its needed because these two contract files are used while initilaizing ethereum_wrapper as part of sdk.

Reply:
@rranjan3 Please remove this dependency in a separate PR